src/test/java/webdriverFramework/ScreenshotTaker.java

```diff
@@ -0,0 +1,35 @@
+package webdriverFramework;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/**
+ * Created by roman.pipchenko on 1/24/2017.
+ */public class ScreenshotTaker extends TestWatcher {
+    private final TakesScreenshot takesScreenshot;
+
+    public ScreenshotTaker(TakesScreenshot takesScreenshot) {
+        this.takesScreenshot = takesScreenshot;
+    }
+
+    @Override
+    protected void failed(Throwable e, Description description) {
+        File screenshot = new File("target" + File.separator +  description.getClassName() + "-" + description.getMethodName() + new SimpleDateFormat("ddHHmmss").format(new Date()) + ".png");
+
+        try {
+            FileUtils.moveFile(takesScreenshot.getScreenshotAs(OutputType.FILE), screenshot);
+        } catch (IOException e1) {
+            throw new RuntimeException(e1);
+        }
+
+        System.err.println("taken " + screenshot);
```

Comment:
Better to avoid System.err or System.out streams. It's too strict. You will not have any way to stream log to any other source or multiply sources. Take a look, for example, into Log4J or java logging. One of these will be preferable for you. When implement it.